File: src/commonMain/kotlin/com/scoreo/application/GetGameTypesUseCase.kt
package com.scoreo.application

import com.scoreo.domain.model.GameType
import com.scoreo.domain.port.GameTypeRepository

class GetGameTypesUseCase(private val repository: GameTypeRepository) {
    operator fun invoke(includeInactive: Boolean = false): List<GameType> = repository.getAll(includeInactive = includeInactive)
}

File: src/application/getGameTypesUseCase.ts
import type { GameType } from '../domain/model/gameType'
import type { GameTypeRepository } from '../domain/port/gameTypeRepository'

export class GetGameTypesUseCase {
  constructor(private readonly repository: GameTypeRepository) {}

  invoke(includeInactive = false): GameType[] {
    return this.repository.getAll(includeInactive)
  }
}
